cmd/backup/script.go

```diff
@@ -421,21 +500,55 @@ func (s *script) stopContainers() (func() error, error) {
 			}
 		}
 
-		if len(restartErrors) != 0 {
+		var scaleUpErrors []error
+		if isDockerSwarm {
+			for _, service := range servicesToScaleDown {
```

Comment:
Shouldn't this be `scaledDownServices`? If a the backup tried to scale a service down, but it ran into an error, what will be the state of it? If the PreviousSpec is not updated, we might make unintentional change to Replica count

Reply:
> If a the backup tried to scale a service down, but it ran into an error, what will be the state of it? 

The idea was to always scale back up services, even if scaling down might have failed (this is also happening for containers), in order to try the very best to restore the previous state. If scaling down failed and we try to update a service to its current state, nothing bad will happen anyways.

You're right about how relying on PreviousSpec only is maybe a bit too risky. I'll think about how to make this a bit more robust.

Reply:
That a good reasoning and then consistent with containers. I am not sure what happens if scale down fails and then we try to scale up. Either way creates an error that should be looked at by the user.